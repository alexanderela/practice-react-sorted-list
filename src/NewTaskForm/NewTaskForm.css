.NewTaskForm {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: space-between;
	width: 30rem;
	height: 6rem;
}

input {
	width: 5rem;
}

button {
	width: 5rem;
}