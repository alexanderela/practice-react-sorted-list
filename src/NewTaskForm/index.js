import React, { Component } from 'react';
import './NewTaskForm.css';

class NewTaskForm extends Component {
	constructor(props) {
		super(props)
		this.state = {
			task: '',
			error: ''
		}
	}

	handleInput = (e) => {
		const task = e.target.value;
		this.setState({ task })
	}

	handleSubmit = (e) => {
		e.preventDefault()
		const task = this.state.task.trim()
		if(task.length) {
			this.props.addTask(this.state)
			this.setState({ task: '', error: '' })
		} else {
			this.setState({ error: 'Please enter a task that is at least 1 letter long.' })
		}
	}

	render() {
		const { task, error } = this.state

		return (
			<form 
				className='NewTaskForm'
				onSubmit={this.handleSubmit}
			>
				<input 
					placeholder='Enter new task'
					name='task'
					value={task}
					onChange={this.handleInput}
				/>
				<button>Submit task</button>
				<h3 className='form-error'>{error.length ? error : ''}</h3>
			</form>
		)
	}
}

export default NewTaskForm;