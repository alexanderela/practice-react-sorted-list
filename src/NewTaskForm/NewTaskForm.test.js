import React from 'react';
import ReactDOM from 'react-dom';
import NewTaskForm from './';
import { shallow } from 'enzyme';

describe('NewTaskForm', () => {
	let wrapper;
	let mockEvent;
	let expected;
	let mockFunc;

	beforeEach(() => {
		mockFunc = jest.fn()
		wrapper = shallow(<NewTaskForm addTask={mockFunc}/>)
		mockEvent = {
			target: {
				value: 'Example typed task'
			},
			preventDefault: jest.fn()
		}
	})

	it('should render like snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	describe('handleInput', () => {
		it('should setState when user inputs task', () => {
			expected = 'Example typed task'
			wrapper.instance().handleInput(mockEvent)
			expect(wrapper.state().task).toEqual(expected)
			expect(wrapper.state().error).toEqual('')
		})
	})

	describe('handleSubmit', () => {		
		it('should invoke addTask if input has length', () => {
			mockEvent = {
				target: {
					value: 'Input text has length'
				},
				preventDefault: jest.fn()
			}		
			wrapper.instance().handleSubmit(mockEvent)
			expect(wrapper.instance().props.addTask).toHaveBeenCalled()
		})
		
		it('should clear input if input has length', () => {})
		
		it('should set error message to state if input has no length', () => {})
	})

})