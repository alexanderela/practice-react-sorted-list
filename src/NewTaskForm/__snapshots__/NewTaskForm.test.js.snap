// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTaskForm should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewTaskForm
    addTask={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="task"
          onChange={[Function]}
          placeholder="Enter new task"
          value=""
        />,
        <button>
          Submit task
        </button>,
        <h3
          className="form-error"
        >
          
        </h3>,
      ],
      "className": "NewTaskForm",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "task",
          "onChange": [Function],
          "placeholder": "Enter new task",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit task",
        },
        "ref": null,
        "rendered": "Submit task",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "form-error",
        },
        "ref": null,
        "rendered": "",
        "type": "h3",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="task"
            onChange={[Function]}
            placeholder="Enter new task"
            value=""
          />,
          <button>
            Submit task
          </button>,
          <h3
            className="form-error"
          >
            
          </h3>,
        ],
        "className": "NewTaskForm",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "task",
            "onChange": [Function],
            "placeholder": "Enter new task",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit task",
          },
          "ref": null,
          "rendered": "Submit task",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "form-error",
          },
          "ref": null,
          "rendered": "",
          "type": "h3",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
